test/OpenFeature.Contrib.Providers.GOFeatureFlag.Test/GoFeatureFlagProviderTest.cs

```diff
@@ -0,0 +1,443 @@
+        mockHttp.When(prefixEval + "double_key" + suffixEval).Respond(mediaType,
+            "{\"trackEvents\":true,\"variationType\":\"True\",\"failed\":false,\"version\":\"\",\"reason\":\"TARGETING_MATCH\",\"errorCode\":\"\",\"value\":100.25}");
+        mockHttp.When(prefixEval + "flag_not_found" + suffixEval).Respond(mediaType,
+            "{\"trackEvents\":true,\"variationType\":\"SdkDefault\",\"failed\":true,\"version\":\"\",\"reason\":\"ERROR\",\"errorCode\":\"FLAG_NOT_FOUND\",\"value\":\"false\"}");
+        mockHttp.When(prefixEval + "integer_key" + suffixEval).Respond(mediaType,
+            "{\"trackEvents\":true,\"variationType\":\"True\",\"failed\":false,\"version\":\"\",\"reason\":\"TARGETING_MATCH\",\"errorCode\":\"\",\"value\":100}");
+        mockHttp.When(prefixEval + "list_key" + suffixEval).Respond(mediaType,
+            "{\"trackEvents\":true,\"variationType\":\"True\",\"failed\":false,\"version\":\"\",\"reason\":\"TARGETING_MATCH\",\"errorCode\":\"\",\"value\":[\"test\",\"test1\",\"test2\",\"false\",\"test3\"]}");
+        mockHttp.When(prefixEval + "object_key" + suffixEval).Respond(mediaType,
+            "{\"trackEvents\":true,\"variationType\":\"True\",\"failed\":false,\"version\":\"\",\"reason\":\"TARGETING_MATCH\",\"errorCode\":\"\",\"value\":{\"test\":\"test1\",\"test2\":false,\"test3\":123.3,\"test4\":1,\"test5\":null}}");
+        mockHttp.When(prefixEval + "string_key" + suffixEval).Respond(mediaType,
+            "{\"trackEvents\":true,\"variationType\":\"True\",\"failed\":false,\"version\":\"\",\"reason\":\"TARGETING_MATCH\",\"errorCode\":\"\",\"value\":\"CC0000\"}");
+        mockHttp.When(prefixEval + "unknown_reason" + suffixEval).Respond(mediaType,
+            "{\"trackEvents\":true,\"variationType\":\"True\",\"failed\":false,\"version\":\"\",\"reason\":\"CUSTOM_REASON\",\"errorCode\":\"\",\"value\":true}");
+
+        return mockHttp;
+    }
+
+    private static EvaluationContext InitDefaultEvaluationCtx()
+    {
+        return EvaluationContext.Builder()
+            .Set("targetingKey", "d45e303a-38c2-11ed-a261-0242ac120002")
+            .Set("email", "john.doe@gofeatureflag.org")
+            .Set("firstname", "john")
+            .Set("lastname", "doe")
+            .Set("anonymous", false)
+            .Set("professional", true)
+            .Set("rate", 3.14)
+            .Set("age", 30)
+            .Set("company_info", new Value(new Structure(new Dictionary<string, Value>
+            {
+                { "name", new Value("my_company") },
+                { "size", new Value(120) }
+            })))
+            .Set("labels", new Value(new List<Value>
+            {
+                new("pro"),
+                new("beta")
+            }))
+            .Build();
+    }
+
+
+    [Fact]
+    private void getMetadata_validate_name()
+    {
+        var provider = new GoFeatureFlagProvider(new GoFeatureFlagProviderOptions

```

Comment:
I think I'd prefer to see "integration" tests as well, where an OpenFeature client is created, but I understand that's more difficult and I think this is sufficent.

Reply:
I have changed the test to use the SDK and not the provider only.
It was actually pretty easy to do it.